### ISSUE TITLE
Add OnStarted, OnStopped events

### ISSUE THREAD

**Issue by Skeatwin** [Eastrall/Ether.Network#109]
Hi,

To confirm the good creation/deletion of listening socket, is it possible to create event for that ?

Thank by advance.

**Comment by johmarjac** [comment]
An event would not be the correct way imo, however there is the Initialize() method which can be overriden by your class inheriting NetServer. But there is not Uninitialize() method that is right. Also the Initialize() method is called before the server socket is actually being created so it might be a good idea to create those two methods OnStart and OnStop which should be virtual and therefore be able to be overriden in your class subclassing NetServer.

**Comment by Skeatwin** [comment]
Yes, we need to know if socket is correctly listening. OnStart/OnStop are more appropriated.

**Comment by Eastrall** [comment]
I guess it would be a nice feature to implement. I add it to the V3.1.0 roadmap.
